src/llama-kv-cache.cpp

```diff
@@ -1898,6 +1916,11 @@ llama_kv_cache_recurrent::llama_kv_cache_recurrent(
     v_l.reserve(n_layer);
 
     for (int i = 0; i < n_layer; i++) {
+        if (filter && !filter(i)) {
```

Comment:
This is buggy because `push_back` is used below, so later when we index directly into the given per-layer tensor vectors, the final layers will be out-of-bounds reads.